### ISSUE TITLE
Creating an ASG with public subnets fails.

### ISSUE THREAD

**Issue by vaneek** [aws/aws-cdk#3410]
**Note: for support questions, please first reference our [documentation](https://docs.aws.amazon.com/cdk/api/latest), then use [Stackoverflow](https://stackoverflow.com/questions/ask?tags=aws-cdk)**. This repository's issues are intended for feature requests and bug reports.

* **I'm submitting a ...**
  - [X] :beetle: bug report
  - [ ] :rocket: feature request
  - [x] :books: construct library gap
  - [ ] :phone: security issue or vulnerability => Please see [policy](https://github.com/aws/aws-cdk/security/policy)
  - [ ] :question: support request => Please see note at the top of this template.

* **What is the current behavior?**
*If the current behavior is a :beetle:bug:beetle:: Please provide the steps to reproduce*
1. Creating an ASG using an imported VPC with only public subnets fails with the following error message,
```
jsii.errors.JSIIError: There are no 'Private' subnets in this VPC. Use a different VPC subnet selection.
```
Code to create stack,
```
class ASGStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        vpc = ec2.Vpc.from_lookup(self, "VPC", is_default=True)
        
        asg = autoscaling.AutoScalingGroup(
            self, "ASG",
            vpc=vpc,
            vpc_subnets=vpc.select_subnets(subnet_type=ec2.SubnetType.PUBLIC, one_per_az=True),
            instance_type=ec2.InstanceType.of(
                ec2.InstanceClass.BURSTABLE3, ec2.InstanceSize.LARGE
            ),
            machine_image=ec2.GenericLinuxImage({
                'eu-west-2': 'ami-1234,
                'eu-west-1': 'ami-5678,
            }),
        )
```

2. Using a VPC created by the same stack ASG construct only uses PRIVATE subnets even when explicitly selecting PUBLIC subnets.
```
class ASGStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)
        
        vpc = ec2.Vpc(self, "VPC")
        
        asg = autoscaling.AutoScalingGroup(
            self, "ASG",
            vpc=vpc,
            vpc_subnets=vpc.select_subnets(subnet_type=ec2.SubnetType.PUBLIC, one_per_az=True),
            instance_type=ec2.InstanceType.of(
                ec2.InstanceClass.BURSTABLE3, ec2.InstanceSize.LARGE
            ),
            machine_image=ec2.GenericLinuxImage({
                'eu-west-2': 'ami-1234,
                'eu-west-1': 'ami-5678,
            }),
        )
```

Snippet from generated cloudformation template,
```
        "VPCZoneIdentifier": [
          {
            "Ref": "VPCPrivateSubnet1Subnet8BCA10E0"
          },
          {
            "Ref": "VPCPrivateSubnet2SubnetCFCDAA7A"
          },
          {
            "Ref": "VPCPrivateSubnet3Subnet3EDCD457"
          }
        ]
```

* **What is the expected behavior (or behavior of feature suggested)?**
Subnet placement to use selected subnets.

* **What is the motivation / use case for changing the behavior or adding this feature?**

* **Please tell us about your environment:**

ℹ️ CDK Version: 1.1.0 (build 1a11e96)
ℹ️ AWS environment variables:
  - AWS_CLOUDWATCH_HOME = /opt/aws/apitools/mon
  - AWS_PATH = /opt/aws
  - AWS_AUTO_SCALING_HOME = /opt/aws/apitools/as
  - AWS_ELB_HOME = /opt/aws/apitools/elb
ℹ️ No CDK environment variables

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. associated pull-request, stackoverflow, gitter, etc)

**Comment by vaneek** [comment]
Explicitely tagging a subnet in an imported VPC with ```aws-cdk:subnet-type``` as ```Public``` also does not work as in example 1. Falsely tagging as ```Public``` and using default subnet selection succeeds. It appears the logic has gotten off tracks somewhere as it is only possible to create an ASG with ```Private``` subnets.

**Comment by rix0rrr** [comment]
Can you check whether this has gotten fixed by https://github.com/aws/aws-cdk/pull/3784 ?

**Comment by SomayaB** [comment]
> Can you check whether this has gotten fixed by #3784 ?

@vaneek

**Comment by rix0rrr** [comment]
If not there it will be fixed by https://github.com/aws/aws-cdk/pull/4668. Closing.
